File: android/src/main/kotlin/com/cloudwebrtc/webrtc/utils/LocalTrackIdGenerator.kt
package com.cloudwebrtc.webrtc.utils

/**
 * Generates unique IDs for local media tracks.
 */
object LocalTrackIdGenerator {
    /**
     * Last created unique ID.
     */
    private var lastId: Int = 0

    /**
     * @return  New unique ID for the local media track.
     */
    fun nextId(): String {
        return "local-" + lastId++.toString()
    }
}

File: ios/Classes/utils/LocalTrackIdGenerator.swift
/// Unique ID generator for the all local `MediaStreamTrackProxy`s.
class LocalTrackIdGenerator {
  /// Singleton instance of `LocalTrackIdGenerator`.
  static let shared: LocalTrackIdGenerator = .init()

  /// Last generated track ID.
  private var lastId: Int = 0

  /// Returns a new local `MediaStreamTrackProxy` unique ID.
  func nextId() -> String {
    self.lastId += 1
    return "local-\(self.lastId)"
  }
}
